main.tf

```diff
@@ -29,4 +29,21 @@ module "zone" {
 
   name = var.name
   tags = var.tags
+  vpc  = var.vpc
+}
+
+module "vpc_association" {
+  source = ".//modules/vpc_association"
+  count  = var.vpc != null ? 1 : 0
+
+  vpc_id  = var.vpc.vpc_id
+  zone_id = var.zone_id
+}
+
+module "zone_association" {
+  source = ".//modules/zone_association"
+  count  = var.vpc != null ? 1 : 0
+
+  vpc_id  = var.vpc.vpc_id
+  zone_id = var.zone_id
 }
```

Comment:
Ok, reading up on this a bit more, and understanding how multiple VPCs can be associated to the zone, and how each VPC _may_ belong to a different account. This complicates things. There's really no proper way to represent that relationship between the VPCs and their provider in the top-level module. So, my suggestion for now is to remove these from the top-level module. Just pass `vpcs` through to the zone module, which works fine when the VPCs all belong to the same account as the zone.

For the cross-account use case, we'll address that in the `vpc_association` module...